vllm/worker/multi_step_model_runner.py

```diff
@@ -271,6 +313,20 @@ def execute_model(
             model_input = self._advance_step(
                 model_input, model_input.cached_outputs[-1].sampler_output)
 
+        output_proc_callback = None
+        if frozen_model_input.use_async_and_multi_step:
+            output_proc_callback = frozen_model_input.async_callback
+            assert output_proc_callback is not None
+            async_callback = functools.partial(
+                self._async_process_outputs,
+                model_input=model_input,
+                output_proc_callback=output_proc_callback)
+
+            frozen_model_input = dataclasses.replace(  # type: ignore
```

Comment:
model_input.frozen_model_input also needs to be updated after replacing this object right?

Reply:
You right, I need to update model_input.frozen_model_input too and this can be done once on first step only. Thanks for catching this!

Reply:
actually it hurts performance for some reason, reverted this for now.